compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -108,9 +108,14 @@ class Compiler {
            new CollectSuperCalls,   // Find classes that are called with super
            new DropInlined,         // Drop Inlined nodes, since backend has no use for them
            new LabelDefs),          // Converts calls to labels to jumps
-      List(new GenBCode)            // Generate JVM bytecode
+      List(checkedIdOverflow(new GenBCode)) // Generate JVM bytecode
     )
 
+  private def checkedIdOverflow(phase: Phase): Phase = {
+    assert(phase.id <= Periods.MaxPossiblePhaseId)
```

Comment:
I don't know if it will ever fail, but that was the check that was there before. Probably at some point this was failing.